app/uk/gov/hmrc/trusts/models/SchemaModels.scala

```diff
@@ -0,0 +1,588 @@
+
+case class AssetMonetaryAmount(assetMonetaryAmount: Long)
+
+object AssetMonetaryAmount {
+  implicit val assetMonetaryAmountFormat: Format[AssetMonetaryAmount] = Json.format[AssetMonetaryAmount]
+}
+
+case class Declaration(name: NameType,
+                       address: AddressType)
+
+object Declaration {
+  implicit val declarationFormat: Format[Declaration] = Json.format[Declaration]
+}
+
+case class Trust(
+                  details: TrustDetailsType,
+                  entities: TrustEntitiesType,
+                  assets: Assets)
+
+object Trust {
+  implicit val trustFormat: Format[Trust] = Json.format[Trust]
+}
+
+case class TrustEntitiesType(naturalPerson: Option[List[NaturalPersonType]],
+                             beneficiary: BeneficiaryType,
+                             deceased: Option[WillType],
+                             leadTrustees: LeadTrusteeType,
+                             trustees: Option[List[TrusteeType]],
+                             protectors: Option[ProtectorsType],
+                             settlors: Option[Settlors])
+object TrustEntitiesType {
+  implicit val trustEntitiesTypeFormat: Format[TrustEntitiesType] = Json.format[TrustEntitiesType]
+}
+
+case class ProtectorsType(protector: Option[List[Protector]],
+                          protectorCompany: Option[List[ProtectorCompany]])
+
+object ProtectorsType {
+  implicit val protectorsTypeFormat: Format[ProtectorsType] = Json.format[ProtectorsType]
+}
+
+case class Protector(name: NameType,
+                     dateOfBirth: Option[DateTime],
+                     identification: Option[IdentificationType])
+
+object Protector {
+  val dateTimePattern = "yyyy-MM-dd"
+
+  implicit val dateFormat = Format[DateTime]( Reads.jodaDateReads(dateTimePattern), Writes.jodaDateWrites(dateTimePattern) )
+
+  implicit val protectorFormat: Format[Protector] = Json.format[Protector]
+}
+
+case class ProtectorCompany(name: String,
+                            identification: IdentificationOrgType)
+
+object ProtectorCompany {
+  implicit val protectorCompanyFormat: Format[ProtectorCompany] = Json.format[ProtectorCompany]
+}
+
+
+
+case class TrusteeType(trusteeInd : Option[TrusteeIndividualType], trusteeOrg : Option[TrusteeOrgType])
+object TrusteeType {
+  implicit val trusteeTypeFormat : Format[TrusteeType] = Json.format[TrusteeType]
+}
+
+case class TrusteeOrgType(name: String,
+                          phoneNumber: Option[String] = None,
+                          email: Option[String] = None,
+                          identification: IdentificationOrgType)
+
+object TrusteeOrgType {
+  implicit val trusteeOrgTypeFormat: Format[TrusteeOrgType] = Json.format[TrusteeOrgType]
+}
+
+case class TrusteeIndividualType(name: NameType,
+                       dateOfBirth: DateTime,
+                       phoneNumber: Option[String],
+                       identification: IdentificationType)
+
+object TrusteeIndividualType {
+  val dateTimePattern = "yyyy-MM-dd"
```

Comment:
This has been defined more than once, can we move this to a constant?

Reply:
Done